cargo-spatial/src/opt.rs

```diff
@@ -0,0 +1,78 @@
+use std::path::PathBuf;
+use structopt::StructOpt;
+
+/// Build, run, and deploy SpatialOS workers written in Rust using Cargo.
+#[derive(StructOpt)]
+#[structopt(name = "cargo-spatial", rename_all = "kebab-case")]
+pub struct Opt {
+    /// Print output in JSON format
+    ///
+    /// Useful when you need to parse the Spatial CLI output.
+    #[structopt(long, short)]
+    pub json_output: bool,
+
+    /// Disable dynamic output elements such as the spinner, progress bars, etc.
+    #[structopt(long, short)]
+    pub no_animation: bool,
+
+    /// Sets the directory log files will be created in
+    ///
+    /// If not specified, this is set to <project_root>/logs when inside a project
+    /// directory and logging is disabled when outside a project directory.
+    #[structopt(parse(from_os_str))]
+    pub log_directory: Option<PathBuf>,
+
+    /// Print verbose logs while running
+    #[structopt(long, short)]
+    pub verbose: bool,
+
+    #[structopt(subcommand)]
+    pub command: Command,
+}
+
+#[derive(StructOpt)]
+#[structopt(rename_all = "kebab-case")]
+pub enum Command {
+    /// Perform code generation from schema files in the project
+    #[structopt(name = "codegen")]
+    Codegen,
+
+    /// Commands for developing and running a local SpatialOS project
+    #[structopt(name = "local")]
+    Local(Local),
+
+    /// Various utilities for generating values used in SpatialOS development
+    #[structopt(name = "generate")]
+    Generate {
+        #[structopt(subcommand)]
+        command: Generate,
+    },
+}
+
+#[derive(StructOpt)]
+#[structopt(rename_all = "kebab-case")]
+pub enum Local {
+    /// Start a SpatialOS simulation locally. Automatically builds workers
+    #[structopt(name = "launch")]
+    Launch(LocalLaunch),
+}
+
+#[derive(StructOpt)]
+#[structopt(rename_all = "kebab-case")]
+pub struct LocalLaunch {
+    /// Don't build workers before launching the local deployment
+    #[structopt(long, short)]
+    pub no_build: bool,
```

Comment:
I'm torn on what the default behaviour of this should be. I.e. - build or no build. Personally, I'd lean toward no rebuild and leave it to the user to decide when to build (as its an expensive operation).

Reply:
I'm inclined to keep the workflow and idioms close to what cargo does for normal Rust projects, and `cargo run` will attempt to build your crate before running it. I really like this workflow since it completely removes the ambiguity of "did I remember to build my code before running it?". In fact, `cargo run` doesn't even have an option for _not_ building your code first (though I think it's reasonable to have that option here, at least for now while we're still figuring out what the workflow looks like). Other than the potential cost of building workers, is there any other reason why you think it would be good to default to not building them?

Reply:
For the workflow where you manually connect workers (~90% of my personal workflow) I wouldn't want the extra build.

Although in that case, I may just default back to `spatial` directly. 

This is probably fine for now, one extension to this would be to read the launch config and determine whether the rebuild is required. I think the ergonomics of this can be improved over time (particularly if you have workers in multiple langauges / build systems) - but its a step in the right direction.

Reply:
Hmmmm, I think it's worth having a separate discussion about the workflow you're used to using and how cargo-spatial can best support it. Since I'm new to using SpatialOS I don't have a good sense of what a "normal" workflow looks like, so I just setup enough functionality to support the basic workflow I was using (i.e. ensure that all workers have been built and then run `spatial local launch` for testing). I'll open an issue to discuss workflow ideas 🙂